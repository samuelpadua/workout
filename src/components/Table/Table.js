import React from 'react'
import styled from 'styled-components'
import { colors } from '../../constants'

const TableStyled = styled.table`
  width: 100%;
  text-align: left;
  border-spacing: 0;
  border-collapse: collapse;

  th, td {
    padding: 10px 0;
  }

  th {
    border-bottom: 1px solid ${colors.gray};
  }

  td {
    border-bottom: 1px solid ${colors.lightGray};
  }
`

function Table (props) {
  return (
    <TableStyled>
      { props.children }
    </TableStyled>
  )
}

export default Table
