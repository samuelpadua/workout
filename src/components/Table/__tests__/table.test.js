
import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Table from '../index'

describe('Table', () => {
  it('should render Table', () => {
    const component = (
      <Table>
        <thead>
          <tr>
            <th>Head</th>
          </tr>
        </thead>
        <tbody>
          <td>
            <th>Body</th>
          </td>
        </tbody>
        <tfoot>
          <td>
            <th>Footer</th>
          </td>
        </tfoot>
      </Table>
    )

    const tree = renderer
      .create(component)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
