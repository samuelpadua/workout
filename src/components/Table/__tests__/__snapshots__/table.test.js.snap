// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render Table 1`] = `
.c0 {
  width: 100%;
  text-align: left;
  border-spacing: 0;
  border-collapse: collapse;
}

.c0 th,
.c0 td {
  padding: 10px 0;
}

.c0 th {
  border-bottom: 1px solid #8C8C8C;
}

.c0 td {
  border-bottom: 1px solid #E5E5E5;
}

<table
  className="c0"
>
  <thead>
    <tr>
      <th>
        Head
      </th>
    </tr>
  </thead>
  <tbody>
    <td>
      <th>
        Body
      </th>
    </td>
  </tbody>
  <tfoot>
    <td>
      <th>
        Footer
      </th>
    </td>
  </tfoot>
</table>
`;
